### PR TITLE
add a manually curated grandpa module

### PR DESCRIPTION
**Pull request by rphmeier**
And de-synchronizes the grandpa authorities from the authorship set.

### CODE REVIEW

File: runtime/src/curated_grandpa.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			system::ensure_root(origin)?;
+			grandpa::Module::<T>::schedule_change(voters, T::BlockNumber::zero())?;
+		}
+
+		fn on_finalise(block_number: T::BlockNumber) {
+			// every so often shuffle the voters and issue a change.
+			let shuffle_period: u64 = Self::shuffle_period().as_();
+			if block_number.as_() % shuffle_period == 0 {
+				let mut seed = system::Module::<T>::random_seed().as_ref().to_vec();
+				seed.extend(b"grandpa_shuffling");
```
Comment (shawntabrizi) [inline runtime/src/curated_grandpa.rs]: Is this to ensure that the same seed value -> hash isn't used across multiple modules?

File: runtime/src/curated_grandpa.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				let mut seed = BlakeTwo256::hash(&seed);
+
+				let mut voters = grandpa::Module::<T>::grandpa_authorities();
+				let voter_count = voters.len();
+
+				if voter_count == 0 { return }
+
+				for i in 0..(voter_count - 1) {
```
Comment (shawntabrizi) [inline runtime/src/curated_grandpa.rs]: How many voters before this computation takes longer than a block period?

Comment (rphmeier) [reply runtime/src/curated_grandpa.rs]: probably more than we can reasonably expect -- the same shuffle is used in the parachains module so we are bounded by that already.

File: runtime/src/curated_grandpa.rs (hunk 1 above)
Comment (gavofyork) [reply runtime/src/curated_grandpa.rs]: i reckon so.

Comment (rphmeier) [reply runtime/src/curated_grandpa.rs]: yup.

**Pull request merged**